docs/explanation/design.md

```diff
@@ -0,0 +1,43 @@
+## Design Philosophy
+
+First and foremost, this is a color scheme. Making colors work
+well takes precedence over other kinds of functionality.
```

Comment:
I like how we are referring to other philosophies but I think I would like to see more content here talking about how _historically_, the plugin kept on growing essentially because no one was there to push back on changes that went against, for example, parts of the UNIX philosophy and how that's made it quite bloated, difficult to maintain, etc. 

Then we can wrap up by saying that we will be trying to align with the philosophies listed below and instead go against what the plugin has historically offered in terms of functionality.